bcgov/mds
# [MDS-4705] MineSpace Incident View and Submit Final Report

## Objective 

[MDS-4705](https://bcmines.atlassian.net/browse/MDS-4705)

_Changes_

- Ported over ScrollSideMenu component/styles from Core
- Implemented fixed side menu and header while navigation down the page(only when in final review state)
    - ![image](https://user-images.githubusercontent.com/13891487/198117777-6b40fd31-9aa8-4fe1-9341-ce0b6a361f5d.png)
- Added status code callout widget to display when in a final review state(top of form)
    - ![image](https://user-images.githubusercontent.com/13891487/198117551-2b72ab2b-36b8-4357-8911-8cea40c5b685.png)
- Added conditional rendering on IncidentPage/Form when in a final review state(not DFT, CLD, IRS)
    - Added Ministry Follow Up section(bottom of form) when in final review state
    - Added Documentation -> Final Report section when form determines it is a dangerous occurence(mine or EMLI)
    - 
![image](https://user-images.githubusercontent.com/13891487/198117945-ad5cc1e5-5a66-4999-9eb1-171c4198025a.png)
- Added conditional validation on reported_to_phone_no/reported_to_email(one must be set at minimum)
- Created new modal for uploading incident documents/final report
- Removed old component files that were misnamed and abandoned

_How to Test_

- Visit the details page of an incident in a final review state(not DFT, CLD, IRS)
    - Verify:
        - All form inputs are disabled
        - Status code widget represents correct status
        - Ministry Follow Up section is displayed
        - Documentation -> Final Report section is displayed if incident is a dangerous occurrence
        - "Add Documentation/Add Final Report" buttons display next to document section titles which open an upload modal
            - Modal supports two different document types(INI, FIN) make sure both get uploaded with correct type(there is a known bug where uploading more documents to an existing incident mutates the document collection, ticket created)
